evmrpc/tracers.go

```diff
@@ -267,10 +273,16 @@ func (api *DebugAPI) TraceBlockByNumber(ctx context.Context, number rpc.BlockNum
 	defer cancel()
 
 	latest := api.ctxProvider(LatestCtxHeight).BlockHeight()
-	if api.maxBlockLookback >= 0 && number.Int64() < latest-api.maxBlockLookback {
+	if api.maxBlockLookback >= 0 && number != rpc.LatestBlockNumber && number != rpc.FinalizedBlockNumber && number.Int64() < latest-api.maxBlockLookback {
 		return nil, fmt.Errorf("block number %d is beyond max lookback of %d", number.Int64(), api.maxBlockLookback)
 	}
 
+	// Verify app state contains history.
+	earliestVersion := api.backend.app.CommitMultiStore().GetEarliestVersion()
+	if earliestVersion > number.Int64() {
```

Comment:
This isn't gonna work since it would still fail if number is latest height right? like -2, -1

Reply:
refactoring into separate helper, will ping when this is ready